### PR TITLE
[CNFT1-2207] Modernized Authentication

### PR DESCRIPTION
**Pull request by adamloup-enquizit**
## Description

Adds configuration to the projects that allow use of OpenID Connect as the authentication mechanism for each of the backend services.

- Configuration added to the `authentication` library to support a standardized OAuth2 Resource Server configuration. 
- Configuration added to the `nbs-gateway` that registers it as an OAuth2 Client
- Default configuration added to `modernization-api` and `question-bank` to act as an OAuth2 Resource Server

## Tickets

* [CNFT1-2207](https://cdc-nbs.atlassian.net/browse/CNFT1-2207)

## Checklist before requesting a review
- [ ] PR focuses on a single story
- [ ] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [ ] All new functions/classes/components reasonably small
- [ ] Functions/classes/components focused on one responsibility
- [ ] Code easy to understand and modify (clarity over concise/clever)
- [ ] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [ ] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT1-2207]: https://cdc-nbs.atlassian.net/browse/CNFT1-2207?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/nbs-gateway/src/test/java/gov/cdc/nbs/security/oidc/NBS6LoginRouteLocatorConfigurationTest.java
Comment (stevegsa) [inline apps/nbs-gateway/src/test/java/gov/cdc/nbs/security/oidc/NBS6LoginRouteLocatorConfigurationTest.java:1]: nice!

File: apps/nbs-gateway/src/test/java/gov/cdc/nbs/security/oidc/NBS6LoginRouteLocatorConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import static org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.springSecurity;
+
+@SpringBootTest(
+    classes = {GatewayApplication.class, NBS6LoginRouteLocatorConfiguration.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {
+        "nbs.gateway.classic=http://localhost:10000"
+    }
+)
+class NBS6LoginRouteLocatorConfigurationTest {
```
Comment (stevegsa) [inline apps/nbs-gateway/src/test/java/gov/cdc/nbs/security/oidc/NBS6LoginRouteLocatorConfigurationTest.java:28]: nice!

Comment (adamloup-enquizit) [reply apps/nbs-gateway/src/test/java/gov/cdc/nbs/security/oidc/NBS6LoginRouteLocatorConfigurationTest.java:28]: This is as far as i'll be able to go without setting up a Keycloak containerin the test

**Review by stevegsa — APPROVED** [review]
Excellent work!

**Review by mpeels — APPROVED** [review]

**Pull request merged**